test/end-to-end/shared/components/bhTagSelect.js

```diff
@@ -4,6 +4,16 @@ module.exports = {
   selector : '[bh-tag-select]',
   set      : async function set(tags) {
     const root = $(this.selector);
-    await FU.typeaheadAppended('$ctrl.$ctrl.tagUuids', tags, root);
+
+    if (Array.isArray(tags)) {
+      const stack = [];
+      for (let index = 0; index < tags.length; index++) {
+        const tag = tags[index];
+        stack.push(FU.uiSelect('$ctrl.tagUuids', tag, root));
+      }
+      await Promise.all(stack);
```

Comment:
Speaking from experience, you should call `await` inside the for loop and remove `Promise.all`.  What you are doing here is an excellent performance increase by making the clicks _parallel_, but will cause random failures in the test since a user cannot actually select things in parallel.  We had all sorts of random crashes with the bhAccountSelect because of this.

Another way to think about it: the E2E test should emulate a user's behavior.  They are not going to click all the tags at once - they will go one tag at a time and click it.  Removing `Promise.all()` and using `await` in the loop will better simulate that experience.